WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -629,6 +651,56 @@ private extension OrderDetailsViewController {
         present(navigationController, animated: true, completion: nil)
     }
 
+    func trashOrderTapped() {
+        ServiceLocator.analytics.track(.orderDetailTrashButtonTapped)
+
+        let alertController = UIAlertController(title: Localization.Alert.orderTrashConfirmationTitle,
+                                                message: Localization.Alert.orderTrashConfirmationMessage,
+                                                preferredStyle: .alert)
+        let cancel = UIAlertAction(title: Localization.Alert.orderTrashConfirmationCancelButton,
+                                   style: .cancel) { (action) in
+        }
+        let confirm = UIAlertAction(title: Localization.Alert.orderTrashConfirmationConfirmButton,
+                                    style: .default) { [weak self] (action) in
+            self?.trashOrderAction()
+        }
+        alertController.addAction(cancel)
+        alertController.addAction(confirm)
+        present(alertController, animated: true)
+    }
+
+    func trashOrderAction() {
+        let order = viewModel.order
+        viewModel.trashOrder { [weak self] result in
+            switch result {
+            case .success:
+                NotificationCenter.default.post(name: .ordersBadgeReloadRequired, object: nil)
+                self?.displayOrderTrashUndoNotice {
```

Comment:
I don't see the undo notice being displayed in iPad simulator. Is this expected? 

https://github.com/woocommerce/woocommerce-ios/assets/524475/ec09eb45-fca9-4a99-af95-67272ce054bc